baselinemarkers/get_marker_genes_rnaseq.R

```diff
@@ -0,0 +1,395 @@
+              by = "GENE_ID")
+
+
+
+head(final_df)
+dim(final_df)
+
+# Ensure EXPRESSION is numeric
+final_df$EXPRESSION <- NA_real_
+
+# Loop over rows to update EXPRESSION based on matching GENE_ID and GROUP
+for (i in seq_len(nrow(final_df)) ) {
+    gene_id <- final_df$GENE_ID[i]
+    group <- final_df$GROUP[i]
+    
+    # Check if gene exists in expression_decorated and group column exists
+    row_match <- which(expression_decorated$GeneID == gene_id)
+    if (length(row_match) == 1 && group %in% colnames(expression_decorated)) {
+        expr_string <- expression_decorated[row_match, group]
+        expr_values <- as.numeric(unlist(strsplit(expr_string, ",")))
+        
+        # Take 3rd value if it exists
+        if (length(expr_values) >= 3) {
+            final_df$EXPRESSION[i] <- expr_values[3]
+        }
+    }
+}
+
+
+
+###############################################################
+# apply cutoff of gene expression and reorder ranking
+###############################################################
+
+# Identify genes where any row has RANKING != -1 AND EXPRESSION <= EXPRESSION_CUTOFF
+genes_to_remove <- final_df %>%
+    filter(RANKING != -1, EXPRESSION <= EXPRESSION_CUTOFF) %>%
+    distinct(GENE_ID) %>%
+    pull(GENE_ID)
+
+# remove all rows for those genes
+filtered_df <- final_df %>%
+    filter(!GENE_ID %in% genes_to_remove)
+
+# reassign RANKING within each GROUP_NAME for remaining valid rows
+# Re-rank genes within each group by ascending specificity, then descending expression, then gene ID (if RANKING != -1)
+
+filtered_df <- filtered_df %>%
+  group_by(GROUP_NAME) %>%
+  arrange(SPECIFICITY_SCORE, desc(EXPRESSION), GENE_ID, .by_group = TRUE) %>%
+  mutate(
+    new_rank = NA_integer_,  # initialize
+    new_rank = replace(new_rank, RANKING != -1, row_number()),  # assign ranks where applicable
+    RANKING = if_else(is.na(new_rank), -1L, new_rank)
+  ) %>%
+  select(-new_rank) %>%
+  ungroup()
+
+
+
+# get unique genes ordered by RANKING within each group using reframe()
+ordered_genes <- filtered_df %>%
+    filter(RANKING > 0) %>%
+    group_by(GROUP_NAME) %>%
+    arrange(RANKING, .by_group = TRUE) %>%
+    reframe(GENE_ID = unique(GENE_ID)) %>%
+    pull(GENE_ID)
+
+# Use those ordered genes to sort the full dataframe
+filtered_df_sorted <- filtered_df %>%
+    filter(GENE_ID %in% ordered_genes) %>%
+    mutate(GENE_ID = factor(GENE_ID, levels = ordered_genes)) %>%
+    arrange(GENE_ID, GROUP_NAME)
+
+# replace -1 with NULL for sql loading
+filtered_df_sorted <- filtered_df_sorted %>%
+    mutate(
+        RANKING = ifelse(RANKING == -1, "NULL", as.character(RANKING)),
+        SPECIFICITY_SCORE = ifelse(SPECIFICITY_SCORE == -1, "NULL", as.character(SPECIFICITY_SCORE))
+    )
+
+colnames(filtered_df_sorted) <- c("experiment_accession", "assay_id", "assay", "gene_id", "specificity_score", "marker_gene_rank", "expression_unit", "number_assays", "expression_level", "gene_name")
```

Comment:
```suggestion
colnames(filtered_df_sorted) <- c("experiment_accession", "assay", "gene_id", "specificity_score", "marker_gene_rank", "expression_unit", "number_assays", "expression_level", "gene_name", "assay_id")
```

moving `assay_id` to end to match schema